crates/aranya-client/src/aqc/api.rs

```diff
@@ -0,0 +1,481 @@
+        let (client, sender) = AqcClient::new(tls_client_provider, client_keys)?;
+
+        let server_addr = aqc_addr
+            .to_string()
+            .parse::<SocketAddr>()
+            .map_err(AqcError::AddrParse)?;
+        // Use the rustls server provider
+        let server = Server::builder()
+            .with_tls(tls_server_provider) // Use the wrapped server config
+            .map_err(|e| AqcError::TlsConfig(e.to_string()))?
+            .with_io(server_addr)
+            .map_err(|e| AqcError::IoConfig(e.to_string()))?
+            .with_congestion_controller(Bbr::default())
+            .map_err(|e| AqcError::CongestionConfig(e.to_string()))?
+            .start()
+            .map_err(|e| AqcError::ServerStart(e.to_string()))?;
+        Ok((Self { client }, server, sender, identity_rx))
+    }
+
+    /// Returns the local address that AQC is bound to.
+    pub async fn local_addr(&self) -> Result<SocketAddr, AqcError> {
+        Ok(self.client.local_addr().await?)
+    }
+
+    /// Creates a bidirectional AQC channel with a peer.
+    pub async fn create_bidirectional_channel(
+        &mut self,
+        peer_addr: SocketAddr,
+        label_id: LabelId,
+        id: BidiChannelId,
+        psk: PresharedKey,
+    ) -> Result<AqcBidirectionalChannel, AqcError> {
+        self.client
+            .create_bidirectional_channel(peer_addr, label_id, id, psk)
+            .await
+            .map_err(AqcError::Other)
+    }
+
+    /// Creates a unidirectional AQC channel with a peer.
+    pub async fn create_unidirectional_channel(
+        &mut self,
+        peer_addr: SocketAddr,
+        label_id: LabelId,
+        id: UniChannelId,
+        psk: PresharedKey,
+    ) -> Result<AqcSenderChannel, AqcError> {
+        self.client
+            .create_unidirectional_channel(peer_addr, label_id, id, psk)
+            .await
+            .map_err(AqcError::Other)
+    }
+
+    /// Receives a channel.
+    pub async fn receive_channel(&mut self) -> Option<AqcChannelType> {
+        self.client.receive_channel().await
+    }
+}
+
+#[derive(Debug)]
+struct ServerPresharedKeys {
+    keys: HashMap<Vec<u8>, Arc<PresharedKey>>,
+    // Optional sender to report the selected identity
+    identity_sender: mpsc::Sender<Vec<u8>>,
+}
+
+impl ServerPresharedKeys {
+    fn new() -> (Self, mpsc::Receiver<Vec<u8>>) {
+        // Create the mpsc channel for PSK identities
+        let (identity_tx, identity_rx) = mpsc::channel::<Vec<u8>>(1); // Buffer size 1 is likely sufficient
+
+        (
+            Self {
+                keys: HashMap::new(),
+                identity_sender: identity_tx,
+            },
+            identity_rx,
+        )
+    }
+
+    fn insert(&mut self, psk: PresharedKey) {
+        let identity = psk.identity().to_vec();
+        if self.keys.insert(identity.clone(), Arc::new(psk)).is_some() {
+            error!("Duplicate PSK identity inserted: {:?}", identity);
+        }
```

Comment:
Use the entry method so we don't have to clone here

```suggestion
        use std::collections::hash_map::Entry; // This should be located with other imports
        match self.keys.entry(identity) {
            Entry::Vacant(e) => {
                e.insert(Arc::new(psk));
            }
            Entry::Occupied(e) => error!("Duplicate PSK identity inserted: {:?}", e.get()),
        }
```